x/bank/keeper/grpc_query.go

```diff
@@ -75,6 +75,48 @@ func (k BaseKeeper) AllBalances(ctx context.Context, req *types.QueryAllBalances
 	return &types.QueryAllBalancesResponse{Balances: balances, Pagination: pageRes}, nil
 }
 
+// SpendableBalances implements a gRPC query handler for retrieving an account's
+// spendable balances.
+func (k BaseKeeper) SpendableBalances(ctx context.Context, req *types.QuerySpendableBalancesRequest) (*types.QuerySpendableBalancesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	if req.Address == "" {
+		return nil, status.Error(codes.InvalidArgument, "address cannot be empty")
+	}
+
+	addr, err := sdk.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid address: %s", err.Error())
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	balances := sdk.NewCoins()
+	accountStore := k.getAccountStore(sdkCtx, addr)
+
+	pageRes, err := query.Paginate(accountStore, req.Pagination, func(key, value []byte) error {
+		var amount sdk.Int
+		if err := amount.Unmarshal(value); err != nil {
+			return err
+		}
+		balances = append(balances, sdk.NewCoin(string(key), amount))
+		return nil
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "paginate: %v", err)
+	}
+
+	result := sdk.NewCoins()
+	spendable := k.SpendableCoins(sdkCtx, addr)
```

Comment:
> Can we paginate on spendable coins rather than doing the join?


No, you cannot paginate on spendable balances because ultimately, you will have to call `LockedCoins` which gets all the users balances anyway. This is why I implemented it this way.

Reply:
oh, this is the limitation of the interface. So pagination doesn't make much sense here.